<?xml version="1.0" encoding="utf-8"?>

<!-- =================================================================================== -->
<!-- |    This is a test URDF with a minimal robot configuration to start the epick    | -->
<!-- |    driver and its controller.                                                   | -->
<!-- =================================================================================== -->

<robot name="MinimalRobot">
  <link name="world"/>
  <joint name="base_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="world"/>
    <child link="base_link"/>
  </joint>
  <link name="base_link">
    <inertial>
      <mass value="0.01"/>
      <origin xyz="0 0 0"/>
      <inertia ixx="0.001" ixy="0.0" ixz="0.0" iyy="0.001" iyz="0.0" izz="0.001"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.2" radius="0.1"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="1" radius="0.1"/>
      </geometry>
    </collision>
  </link>

  <ros2_control name="${name}" type="system">
    <hardware>
      <plugin>epick_driver/EpickGripperHardwareInterface</plugin>

      <!-- Serial connection parameters. -->
      <param name="usb_port">/dev/ttyUSB0</param>
      <param name="baud_rate">9600</param>
      <param name="timeout">500</param>

      <!-- Gripper parameters. -->
      <param name="slave_address">0x9</param>
      <param name="mode">AutomaticMode</param>
      <param name="max_vacuum_pressure">-100</param>
      <param name="min_vacuum_pressure">-10</param>
      <param name="gripper_timeout">500</param>

      <!-- Set use_dummy to true to connect to a dummy driver. -->
      <param name="use_dummy">true</param>

    </hardware>

    <gpio name="gripper">
      <command_interface name="regulate"/>
      <state_interface name="object_detection_status"/>
    </gpio>

  </ros2_control>

</robot>
"
